test/functorch/common_utils.py

```diff
@@ -26,8 +26,9 @@ def loop(op, in_dims, out_dim, batch_size, *batched_args, **kwarg_values):
     out_spec = None
     for idx in range(batch_size):
         flat_args, args_spec = pytree.tree_flatten(batched_args)
+        args_spec_with_tuple = args_spec.replace_types({torch.Size: tuple})
```

Comment:
> As it turns out, vmap_impl also depends on this quite unusual behaviour - Hyrum's law at work. We fix that by allowing special-case equivalence between tuple and Size for the vmap in_dim case.

Do you have an example of code that worked before that now needs this change to continue working?

Reply:
I was planning to use `tree_map` for mapping some fake tensor values. But I ran into errors saying tuple does not implement `numel`. 

While I decided not to use `tree_map` eventually, I don't think one should expect this behaviour generally when one doesn't do anything funny to the Size ints.

Reply:
> allowing special-case equivalence between tuple and Size for the vmap in_dim case.

As for this part, I think it is necessary in order for vmap to work with its `(1, None)` in_dims

Reply:
sorry, it's still unclear to me what you mean by "vmap relies on it". Do you have a test case that demonstrates the issue?

Reply:
python test/functorch/test_vmap.py -k test_op_has_batch_rule_split_list_args_cpu

Reply:
If you remove the lines where we use the `spec_with_tuple`, it will error. See e.g. https://hud.pytorch.org/pytorch/pytorch/pull/111993?sha=a92ed659722cd895bf23bf447ba912b2a4608d72